src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -388,12 +398,15 @@ protected virtual void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    if (_flushLogsTimer?.Value != null)
+                    if (_flushLogsTimer?.Value is not null)
                     {
                         _flushLogsTimer?.Value?.Dispose();
                     }
 
-                    FlushLogs().GetAwaiter().GetResult();
+                    if (_tableClient is not null)
```

Comment:
It seems like an abuse to reference the field here, when the intent of the TableClient property is to ensure all usages go through the property.

Perhaps Dispose should just set _isEnabled to false, or call DisableService, which would cause FlushLogs to noop?

Reply:
ok